dev/com.ibm.ws.jbatch.misc_fat/util/src/fat/util/JobWaiter.java

```diff
@@ -0,0 +1,219 @@
+ * {@code BatchStatus.FAILED}.
+ *
+ * The full set of finished statuses is
+ * <li>{@code BatchStatus.COMPLETED}
+ * <li>{@code BatchStatus.FAILED}.
+ * <li>{@code BatchStatus.STOPPED}.
+ * <li>{@code BatchStatus.ABANDONED}.
+ *
+ * It also waits for job exitStatus to be non-null
+ */
+public class JobWaiter {
+
+    private final static Logger logger = Logger.getLogger("test");
+
+    // Full list:
+    //public enum BatchStatus {STARTING, STARTED, STOPPING, STOPPED, FAILED, COMPLETED, ABANDONED }
+    private static Set<BatchStatus> finishedStatuses = new HashSet<BatchStatus>();
+    static {
+        finishedStatuses.add(BatchStatus.STOPPED);
+        finishedStatuses.add(BatchStatus.FAILED);
+        finishedStatuses.add(BatchStatus.COMPLETED);
+        finishedStatuses.add(BatchStatus.ABANDONED);
+    }
+
+    protected static JobOperator jobOp = BatchRuntime.getJobOperator();
+
+    private final int POLL_INTERVAL = 100; // .1 second
+
+    private long timeout;
+
+    /**
+     * @return Time (milliseconds) to wait for job to finish
+     */
+    public long getTimeout() {
+        return timeout;
+    }
+
+    /**
+     * @param timeout Time (milliseconds) to wait for job to finish
+     */
+    public void setTimeout(long timeout) {
+        this.timeout = timeout;
+    }
+
+    /**
+     * @param timeout Time (milliseconds) to wait for job to finish
+     */
+    public JobWaiter() {
+        this(DFLT_TIMEOUT);
+    }
+
+    /**
+     * @param timeout Time (milliseconds) to wait for job to finish
+     */
+    public JobWaiter(long timeout) {
+        super();
+        this.timeout = timeout;
+    }
+
+    private static long DFLT_TIMEOUT = 10000;
+
+    /**
+     * Wait for {@code JobWaiter#timeout} seconds for BOTH of:
+     *
+     * 1) BatchStatus to be one of: STOPPED ,FAILED , COMPLETED, ABANDONED
+     * AND
+     * 2) exitStatus to be non-null
+     *
+     * Returns JobExecution if it ends in <b>COMPLETED</b> status, otherwise throws
+     * <b>IllegalStateException</b>
+     *
+     * @param jobXMLName
+     * @param jobParameters
+     * @return JobExecution (for successfully completing job)
+     * @throws IllegalStateException
+     */
+    public JobExecution completeNewJob(String jobXMLName, Properties jobParameters) throws IllegalStateException {
+        long executionId = jobOp.start(jobXMLName, jobParameters);
+        JobExecution jobExec = waitForFinish(executionId);
+        if (jobExec.getBatchStatus().equals(BatchStatus.COMPLETED)) {
+            logger.finer("Job " + executionId + " successfully completed.");
+            return jobExec;
+        } else {
+            throw new IllegalStateException("Job " + executionId + " finished with non-completed state: " + jobExec.getBatchStatus());
+        }
+    }
+
+    /**
+     * Wait for {@code JobWaiter#timeout} seconds for BOTH of:
+     *
+     * 1) BatchStatus to be one of: STOPPED ,FAILED , COMPLETED, ABANDONED
```

Comment:
This doc (I know I started it) should do a better job explaining how for restartAttempts > 0, the job must not complete on the first execution, but does not need **exactly**  'restartAttempts ' restarts.